SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp

```diff
@@ -217,32 +231,57 @@ void USpatialClassInfoManager::CreateClassInfoForClass(UClass* Class)
 	}
 }
 
-bool USpatialClassInfoManager::IsSupportedClass(const FString& PathName) const
+void USpatialClassInfoManager::FinishConstructingSubobjectClassInfo(const FString& ClassPath, TSharedRef<FClassInfo>& Info)
 {
-	return SchemaDatabase->ClassPathToSchema.Contains(PathName);
+	for (const auto& DynamicSubobjectData : SchemaDatabase->SubobjectClassPathToSchema[ClassPath].DynamicSubobjectComponents)
+	{
+		// Make a copy of the already made FClassInfo for this dynamic subobject
+		TSharedRef<FClassInfo> DynamicSubobjectInfo = MakeShared<FClassInfo>(Info.Get());
+
+		int32 Offset = DynamicSubobjectData.SchemaComponents[SCHEMA_Data];
+		check(DynamicSubobjectData.SchemaComponents[SCHEMA_Data] != SpatialConstants::INVALID_COMPONENT_ID);
+
+		ForAllSchemaComponentTypes([&](ESchemaComponentType Type)
+		{
+			Worker_ComponentId ComponentId = DynamicSubobjectData.SchemaComponents[Type];
+
+			if (ComponentId != SpatialConstants::INVALID_COMPONENT_ID)
+			{
+				DynamicSubobjectInfo->SchemaComponents[Type] = ComponentId;
+				ComponentToClassInfoMap.Add(ComponentId, DynamicSubobjectInfo);
+				ComponentToOffsetMap.Add(ComponentId, Offset);
+				ComponentToCategoryMap.Add(ComponentId, ESchemaComponentType(Type));
+			}
+		});
+
+		Info->DynamicSubobjectInfo.Add(DynamicSubobjectInfo);
```

Comment:
This was mildly confusing to read. Could we name the local variable something else?